### PR TITLE
gh-99991: improve docs on str.encode and bytes.decode

### PR DESCRIPTION
**Pull request by bizzyvinci**
Update str.encode and byte.decode docs.

* Issue: gh-99991

### CODE REVIEW

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -1625,13 +1625,15 @@ expression support in the :mod:`re` module).
 .. method:: str.encode(encoding="utf-8", errors="strict")
 
    Return an encoded version of the string as a bytes object. Default encoding
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   Return the string encoded to :class:`bytes`.
```

* Following the original issue, keep the summary line in its own paragraph for clarity and readability.
* Per the original issue, properly markup/link the `bytes` object.
* Also, per the general principles of reference type documentation, keep this more concise.

File: Doc/library/stdtypes.rst [hunk 2]
```diff
@@ -1625,13 +1625,15 @@ expression support in the :mod:`re` module).
...
-   is ``'utf-8'``. *errors* may be given to set a different error handling scheme.
-   The default for *errors* is ``'strict'``, meaning that encoding errors raise
-   a :exc:`UnicodeError`. Other possible
-   values are ``'ignore'``, ``'replace'``, ``'xmlcharrefreplace'``,
-   ``'backslashreplace'`` and any other name registered via
-   :func:`codecs.register_error`, see section :ref:`error-handlers`. For a
-   list of possible encodings, see section :ref:`standard-encodings`.
+   is ``'utf-8'``.  And for a list of possible encodings, see section
+   :ref:`standard-encodings`.
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   *encoding* defaults to ``'utf-8'``;
   see :ref:`standard-encodings` for possible values.
```

* Per the original issue, give the *encoding* argument it's own paragraph for readability
* Explicitly state and format the parameter name at the beginning per the docs style guide
* Fix and greatly improve grammar/phrasing
* Be more concise and austere in reference docs

File: Doc/library/stdtypes.rst [hunk 3]
```diff
@@ -1625,13 +1625,15 @@ expression support in the :mod:`re` module).
...
+
+   *errors* may be given to set a different error handling scheme.  The
+   default for *errors* is ``'strict'``, meaning that encoding errors raise
+   a :exc:`UnicodeError`. Other possible values are ``'ignore'``,
+   ``'replace'``, ``'xmlcharrefreplace'``, ``'backslashreplace'`` and any
+   other name registered via :func:`codecs.register_error`, see section
+   :ref:`error-handlers`.
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   *errors* controls how encoding errors are handled.
   If ``'strict'`` (the default), a :exc:`UnicodeError` exception is raised.
   Other possible values are ``'ignore'``,
   ``'replace'``, ``'xmlcharrefreplace'``, ``'backslashreplace'`` and any
   other name registered via :func:`codecs.register_error`.
   See :ref:`error-handlers` for details.
```

* Be more direct, descriptive and conventional with summary line
* Be more concise overall and avoid repetition
* Avoid grammar error (comma splice)

File: Doc/library/stdtypes.rst [hunk 4]
```diff
@@ -1625,13 +1625,15 @@ expression support in the :mod:`re` module).
...
 
    By default, the *errors* argument is not checked for best performances, but
    only used at the first encoding error. Enable the :ref:`Python Development
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]: I was not able to actually make this as suggestion, but this implements the revision from the issue, while still including the key details from the previous version.

```rst
   For performance reasons, the value of *errors* is not checked for validity
   unless an encoding error actually occurs,
   :ref:`Python Development Mode <devmode>` is enabled
   or a :ref:`debug build <debug-build>` is used.
```

File: Doc/library/stdtypes.rst [hunk 5]
```diff
@@ -2760,12 +2762,14 @@ arbitrary binary data.
             bytearray.decode(encoding="utf-8", errors="strict")
 
    Return a string decoded from the given bytes.  Default encoding is
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   Return the bytes decoded to a :class:`str`.
```

As above:
* Following the original issue, keep the summary line in its own paragraph for clarity and readability.
* Per the original issue, properly markup/link the `str` object.
* Also, per the general principles of reference type documentation, keep this more concise.

File: Doc/library/stdtypes.rst [hunk 6]
```diff
@@ -2760,12 +2762,14 @@ arbitrary binary data.
...
-   ``'utf-8'``. *errors* may be given to set a different
-   error handling scheme.  The default for *errors* is ``'strict'``, meaning
-   that encoding errors raise a :exc:`UnicodeError`.  Other possible values are
-   ``'ignore'``, ``'replace'`` and any other name registered via
-   :func:`codecs.register_error`, see section :ref:`error-handlers`. For a
-   list of possible encodings, see section :ref:`standard-encodings`.
+   ``'utf-8'``.  And for a list of possible encodings, see section
+   :ref:`standard-encodings`.
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   *encoding* defaults to ``'utf-8'``;
   see :ref:`standard-encodings` for possible values.
```

As above:
* Per the original issue, give the *encoding* argument it's own paragraph for readability
* Explicitly state and format the parameter name at the beginning per the docs style guide
* Fix and greatly improve grammar/phrasing
* Be more concise and austere in reference docs

File: Doc/library/stdtypes.rst [hunk 7]
```diff
@@ -2760,12 +2762,14 @@ arbitrary binary data.
...
+
+   *errors* may be given to set a different error handling scheme.  The
+   default for *errors* is ``'strict'``, meaning that encoding errors raise a
+   :exc:`UnicodeError`.  Other possible values are ``'ignore'``, ``'replace'``
+   and any other name registered via :func:`codecs.register_error`, see
+   section :ref:`error-handlers`.
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]:
```suggestion
   *errors* controls how decoding errors are handled.
   If ``'strict'`` (the default), a :exc:`UnicodeError` exception is raised.
   Other possible values are ``'ignore'``, ``'replace'``,
   and any other name registered via :func:`codecs.register_error`.
   See :ref:`error-handlers` for details.
```

* Fix "encoding errors" -> "decoding errors"

As above:
* Be more direct, descriptive and conventional with summary line
* Be more concise overall and avoid repetition
* Avoid grammar error (comma splice)

File: Doc/library/stdtypes.rst [hunk 8]
```diff
@@ -2760,12 +2762,14 @@ arbitrary binary data.
...
 
    By default, the *errors* argument is not checked for best performances, but
    only used at the first decoding error. Enable the :ref:`Python Development
```
Comment (CAM-Gerlach) [inline Doc/library/stdtypes.rst]: I was not able to actually make this as suggestion, but this implements the revision from the issue, while still including the key details from the previous version.

```rst
   For performance reasons, the value of *errors* is not checked for validity
   unless a decoding error actually occurs,
   :ref:`Python Development Mode <devmode>` is enabled
   or a :ref:`debug build <debug-build>` is used.
```

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks. I've added some suggestions to fix grammar errors, better implement the original feedback in the issue, follow our conventions and the Diataxis principles, and clarify and improve the phrasing and markup, as well as a couple comments I couldn't leave as suggestions that you'll need to implement manually (or I can implement as a commit on your branch, if you prefer).

Just FYI, in case you're unfamiliar—you can directly apply some or all of my suggestions by going to the `Files changed` tab, clicking `Add to batch` on each suggestion, then clicking `Commit` with a descriptive message.

**Comment by bizzyvinci** [comment]
Thanks for the suggestions and comments @CAM-Gerlach. And yes, I'll prefer you implement a commit on my branch.

**Comment by CAM-Gerlach** [comment]
Done, thanks. I also made a few small refinements to the text of the `versionchanged` and notes of the same functions in line with the other changes, as well as fixed a missing line break that GitHub stripped.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM now, thanks @bizzyvinci and @jayaddison !

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**